agent/src/api/v1/proxy.rs

```diff
@@ -1,52 +1,66 @@
 use anyhow::Result;
 use axum::{
+    body::{Body, HttpBody},
     extract::State,
     http::{Request, StatusCode},
     middleware::Next,
     response::{IntoResponse, Response},
 };
 use http::header::{CONTENT_LENGTH, EXPECT, HOST};
 use mdns_sd::ServiceInfo;
+use std::ops::Deref;
 use utils::{
     errors::ServalError,
     mdns::{discover_service, get_service_instance_id},
 };
 
 use super::*;
-use crate::structures::AppState;
+use crate::structures::{AppState, RunnerState};
 
-pub async fn proxy_unavailable_services<B>(
+pub async fn proxy_unavailable_services(
     State(state): State<AppState>,
-    req: Request<B>,
-    next: Next<B>,
+    mut req: Request<Body>,
+    next: Next<Body>,
 ) -> Result<Response, StatusCode> {
     let path = req.uri().path();
-    if path.starts_with("/v1/storage/") {
-        let state = state.lock().await;
-        if state.storage.is_none() {
-            log::info!(
-                "proxy_unavailable_services intercepting request; path={}",
-                path,
-            );
-
-            let Ok(resp) = proxy_request_to_service(&req, "_serval_storage", &state.instance_id).await else {
-                // Welp, not much we can do
-                return Ok((StatusCode::SERVICE_UNAVAILABLE, "Storage not available").into_response());
-            };
-
-            return Ok(resp);
-        }
+
+    let runner_state = state.lock().await;
```

Comment:
The downside of this refactor is that the middleware now briefly gets a lock on the state for all requests. The upside is that the code for `get_proxy_target_service` is much simpler and more readable.

Feedback welcome.